### ISSUE TITLE
fix `fuzz_calculate_present_value`

### ISSUE THREAD

**Issue by dpaiton** [delvtech/hyperdrive-rs#56]
`fuzz_calculate_present_value` is failing on main. It fails intermittently on CI and consistently if you bump up `FAST_FUZZ_RUNS` to 100k

Example failure:

```
running 1 test
test lp::tests::fuzz_calculate_present_value ... FAILED

failures:

---- lp::tests::fuzz_calculate_present_value stdout ----
thread 'lp::tests::fuzz_calculate_present_value' panicked at crates/hyperdrive-math/src/lp.rs:299:79:
called `Result::unwrap()` on an `Err` value: calculate_max_buy_bonds_out_safe: y = 140766449.986666938237369971 < 140781651.077315479405295767 = optimal_y

Location:
    crates/hyperdrive-math/src/yield_space.rs:266:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    lp::tests::fuzz_calculate_present_value

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 42 filtered out; finished in 13.52s
```

**Comment by sentilesdal** [comment]
this might already be fixed, I'm going to run some tests on this to see if it still flakes.

**Comment by sentilesdal** [comment]
This isn't flaking anymore, fixed in #11 .
